aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -240,6 +270,27 @@ Tensor cat_cuda(TensorList inputs, int64_t dimension) {
   return out;
 }
 
+inline c10::MemoryFormat compute_output_memory_format(const TensorList &inputs) {
+  c10::optional<c10::MemoryFormat> format = c10::nullopt;
+  for (auto &t : inputs) {
+    auto f = t.suggest_memory_format();
+    if (!format.has_value()) {
+      format = f;
+      continue;
+    }
+    if (format.value() == f) {
+      continue;
+    }
+    if (format.value() == c10::MemoryFormat::Contiguous || f == c10::MemoryFormat::Contiguous) {
```

Comment:
nit: this and the following conditional can be unified?

Reply:
fixed